helix-term/src/commands.rs

```diff
@@ -4529,20 +4529,33 @@ fn surround_add(cx: &mut Context) {
         };
         let (view, doc) = current!(cx.editor);
         let selection = doc.selection(view.id);
+        let index = selection.primary_index();
         let (open, close) = surround::get_pair(ch);
 
         let mut changes = Vec::with_capacity(selection.len() * 2);
+        let mut ranges = SmallVec::with_capacity(selection.len());
+
         for range in selection.iter() {
             let mut o = Tendril::new();
             o.push(open);
             let mut c = Tendril::new();
             c.push(close);
             changes.push((range.from(), range.from(), Some(o)));
             changes.push((range.to(), range.to(), Some(c)));
+
+            // Add 2 characters to the range to select them
+            ranges.push(if range.anchor < range.head {
+                Range::new(range.anchor, range.head + 2)
+            } else {
+                Range::new(range.anchor + 2, range.head)
+            });
         }
 
         let transaction = Transaction::change(doc.text(), changes.into_iter());
```

Comment:
Transaction takes a `.with_selection` instead of `doc.set_selection` afterwards

Reply:
Fixed in cb0f7c907ed966961b926798e6d909e24bbb6883